### PR TITLE
ci: Set up GitHub workflows

### PR DESCRIPTION
**Pull request by Gustl22**
- Setup GitHub workflows for
  - Automated testing
  - Automated GitHub releases
  - Preparing for Publishing on Web and Google Play Store
  - Test Pull requests
  - Automated publishing on GitHub pages, but needs to be enabled in settings, also.
- Support for
  - macOS
  - Windows
  - Linux
- Updated:
  - iOS
  - Android
  - Web
- Run Configurations for IntelliJ
- Option to override `cptclient.yaml` via Environment variables from command line args

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -22,13 +23,29 @@ if (flutterVersionName == null) {
...
+        main.java.srcDirs += 'src/main/kotlin'
+    }
 
     defaultConfig {
         applicationId "link.zomtir.cptclient"
+        // You can update the following values to match your application needs.
+        // For more information, see: https://docs.flutter.dev/deployment/android#reviewing-the-gradle-build-configuration.
         minSdkVersion flutter.minSdkVersion
-        minSdkVersion 21
-        targetSdkVersion 34
```
Comment (Gustl22) [inline android/app/build.gradle:31]: Better not set these values as it also sometimes depends on what flutter version you have. They will update the versions regularly.

File: android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -1,45 +1,49 @@
-<manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="link.zomtir.cptclient"
-    android:versionCode="1"
-    android:versionName="0.7">
-
-    <!-- Specify the minimum API level -->
-    <uses-sdk android:minSdkVersion="21" />
```
Comment (Gustl22) [inline android/app/src/main/AndroidManifest.xml:7]: No need to set these here anymore, also the versionName and code is updated automatically.

File: android/app/src/main/java/com/zomtir/sportcpt/MainActivity.java
Comment (Gustl22) [inline android/app/src/main/java/com/zomtir/sportcpt/MainActivity.java:1]: Use kotlin main now.

File: lib/main.dart [hunk 3]
```diff
@@ -8,56 +8,65 @@ import 'package:flutter_gen/gen_l10n/app_localizations.dart';
...
-      appBarTheme: AppBarTheme(
-        backgroundColor: Color.fromARGB(255, 208, 190, 135),
-      ),
-      elevatedButtonTheme: ElevatedButtonThemeData(
-        style: ElevatedButton.styleFrom(
-          padding: EdgeInsets.symmetric(horizontal: 50, vertical: 10),
+  runApp(CptApp());
+}
+
+class CptApp extends StatelessWidget {
```
Comment (Gustl22) [inline lib/main.dart:14]: Extracted main app to be testable.

File: lib/static/environment.dart [hunk 4]
```diff
@@ -0,0 +1,38 @@
+enum Env {
+  serverScheme,
+  serverHost,
+  serverPort;
+
+  String? fromString() {
+    switch (this) {
+      case Env.serverScheme:
+        return const bool.hasEnvironment('SERVER_SCHEME')
+            ? const String.fromEnvironment('SERVER_SCHEME')
```
Comment (Gustl22) [inline lib/static/environment.dart:10]: These need to be `const` in order to work!

File: lib/static/navigation.dart [hunk 5]
```diff
@@ -26,9 +27,9 @@ Future<void> connect() async {
   final configString = await rootBundle.loadString('cptclient.yaml');
   final dynamic configMap = loadYaml(configString);
 
-  html.window.localStorage.putIfAbsent('ServerScheme', () => configMap['ServerScheme']);
-  html.window.localStorage.putIfAbsent('ServerHost', () => configMap['ServerHost']);
-  html.window.localStorage.putIfAbsent('ServerPort', () => configMap['ServerPort']);
+  html.window.localStorage.putIfAbsent('ServerScheme', () => Env.serverScheme.fromString() ?? configMap['ServerScheme']);
+  html.window.localStorage.putIfAbsent('ServerHost', () => Env.serverHost.fromString() ?? configMap['ServerHost']);
+  html.window.localStorage.putIfAbsent('ServerPort', () => Env.serverPort.fromInt()?.toString() ?? configMap['ServerPort']);
```
Comment (Gustl22) [inline lib/static/navigation.dart:32]: `dart-define` take precedence over `cptclient.yaml` as the file is required. But one doesn't need to specify them.

File: android/app/build.gradle [hunk 6]
```diff
@@ -1,5 +1,6 @@
 plugins {
     id "com.android.application"
+    id "kotlin-android"
```
Comment (Zomtir) [inline android/app/build.gradle:3]: Why is Kotlin needed?

File: android/app/src/main/java/com/zomtir/sportcpt/MainActivity.java
Comment (Zomtir) [reply android/app/src/main/java/com/zomtir/sportcpt/MainActivity.java:1]: Does this allow to drop some java dependencies or strictly add Kotlin?

Comment (Gustl22) [reply android/app/src/main/java/com/zomtir/sportcpt/MainActivity.java:1]: Kotlin is dependent on java, so don't think so. But this is the recommended language of Android and Flutter, means if you create a new flutter app, it is served with kotlin now. Most Flutter plugins are also written in kotlin.

File: android/app/build.gradle (hunk 6 above)
Comment (Gustl22) [reply android/app/build.gradle:3]: For Flutter plugins

**Pull request merged**